driver-core/src/main/com/mongodb/connection/netty/NettyStream.java

```diff
@@ -174,6 +174,7 @@ public void writeAsync(final List<ByteBuf> buffers, final AsyncCompletionHandler
             io.netty.buffer.ByteBuf byteBuf = ((NettyByteBuf) cur).asByteBuf();
             composite.addComponent(byteBuf.retain());
             composite.writerIndex(composite.writerIndex() + byteBuf.writerIndex());
+            byteBuf.release();
```

Comment:
If the retain two lines up is removed, the reference count will work out to the same as if it's retained and released